### ISSUE TITLE
 mean read length is outside the graph window size (+/- 10 bases)

### ISSUE THREAD

**Issue by cimendes** [will-rowe/groot#7]
Hi! Due to trimming my read data actually ranged in length from 55 to 235. It is possible for groot to accomodate this? In the meantime I'll try to get groot to work by filtering the reads by size. 
Thanks for the help!

**Comment by will-rowe** [comment]
Hey. I will add in some way to accommodate this kind of read length variability. Thanks for opening the first issue - glad to know people are using GROOT!

**Comment by cimendes** [comment]
Thank you very much! :D For anyone else who is running into this problem, filtering the reads with prinseq before running on groot fixed the problem. :)

**Comment by alexmsalmeida** [comment]
Hi, 

Was wondering if any progress was made in this regard. This functionality would also be extremely useful for me. Currently analysing thousands of metagenomes that have reads ranging from 50 to 300 bp, so this would help me a lot. Many thanks in advance.

Alex

**Comment by will-rowe** [comment]
No progress yet, but I work on it. Sorry for dropping the ball on this one. Maybe some sort of containment approach to seeding the shorter reads, will try it out

**Comment by alexmsalmeida** [comment]
Thanks a lot, looking forward to it!

**Comment by will-rowe** [comment]
Hi - I finally got round to this, sorry for the delay.

I have implemented a containment search to accommodate variable read lengths, see the dev branch (46f47f4e0825708e64735a195569fc7cb4c995a9). This has taken a big rewrite and has been possible thanks to this [cool paper](http://www.vldb.org/pvldb/vol9/p1185-zhu.pdf) and the LSH Ensemble code.

On the test data, I can get 97%  of reads to align correctly (variable read length 50-100 bases). The Jaccard parameter works differently here, it is instead used as a containment threshold (set at 0.5 for this test). I need to write some more tests and docs before release but feel free to try it out for now.
